npm/src/auth/index.ts

```diff
@@ -17,3 +21,33 @@ export const getKeycloakInstance: () => Promise<
     throw "User is not authenticated"
   }
 }
+
+export const refreshOrReturnToken: (
+  keycloak: Keycloak.KeycloakInstance<"native">,
+  tokenMinValidityInSecs?: number
+) => Promise<string> = async (keycloak, tokenMinValidityInSecs = 30) => {
+  if (keycloak.isTokenExpired(tokenMinValidityInSecs)) {
+    await keycloak.updateToken(tokenMinValidityInSecs)
+  }
+  if (keycloak.token) {
+    return keycloak.token
+  } else {
+    throw "Token is expired"
+  }
+}
+
+export const authenticateAndGetIdentityId: (
+  keycloak: Keycloak.KeycloakInstance<"native">
+) => Promise<string> = async keycloak => {
+  const token = await refreshOrReturnToken(keycloak)
+  const credentials = new CognitoIdentityCredentials({
+    IdentityPoolId: TDR_IDENTITY_POOL_ID,
+    Logins: {
+      [TDR_IDENTITY_PROVIDER_NAME]: token
+    }
+  })
+  AWS.config.update({ region: "eu-west-2", credentials })
```

Comment:
Maybe region should be set as env variable?

Reply:
I've set it as a webpack variable but then hard coded it into the package json. I don't think we'll ever need to change it but if we do, I can set it in the Jenkins container.